packages/react/src/Tabs/Tabs.tsx

```diff
@@ -13,37 +13,53 @@ import { TabsPanel } from './TabsPanel'
 import { useKeyboardFocus } from '../common/useKeyboardFocus'
 
 export type TabsProps = {
-  /** The number of the active tab. Corresponds to its `tab` value. */
-  activeTab?: number
+  /** The identifier of the initially active tab. Corresponds to its `tab` value. */
+  activeTab?: string
   /* Provides the id of the activated tab. */
-  onTabChange?: (tabId: number) => void
+  onTabChange?: (tabId: string) => void
 } & PropsWithChildren<HTMLAttributes<HTMLDivElement>>
 
 const TabsRoot = forwardRef(
   ({ activeTab, children, className, onTabChange, ...restProps }: TabsProps, ref: ForwardedRef<HTMLDivElement>) => {
     const tabsId = useId()
     const innerRef = useRef<HTMLDivElement>(null)
-    const [activeTabId, setActiveTabId] = useState(0)
+    const [activeTabId, setActiveTabId] = useState<string>()
 
-    const allTabs = useMemo(() => {
+    const allTabIds = useMemo(() => {
       if (!Array.isArray(children)) return []
-      return children[0].props.children
+
+      // The first child of Tabs should be TabsList
+      const tabsList = children[0]
```

Comment:
Shouldn’t we then query for a class of `.ams-tabs__list`?

Reply:
Using something like `querySelector` would move you out of the React context, no? We're currently traversing React props, not DOM nodes. 

Tbh, I think we should rethink this approach at some point, it's a little brittle.

Reply:
Of course 🤦🏼‍♂️ and yes, agreed.